packages/remix-server-runtime/sessions.ts

```diff
@@ -175,6 +175,7 @@ export interface SessionStorage<Data = SessionData, FlashData = Data> {
    * return a new Session with no data.
    */
   getSession(
+    this: void,
```

Comment:
FWIW I would generally consider using an arrow function to be more idiomatic. `this: void` is much less common in TypeScript code and much less understood by a lot of TS devs. Arrow functions are pretty commonly understood.

Maybe there's a docs issue to be filed on us in typescript-eslint? Or perhaps even a lint rule around preferring one vs. the other? What do you think? 🤔

Reply:
See my other [comment](https://github.com/remix-run/remix/pull/6330#discussion_r1186739382), because just converting to arrow functions won't enforce the documentation contract of those functions being destructured. I'm not sure which way is the right way for Remix, but there is a hidden side-effect of just using arrow functions.

Reply:
Having `this: void` in an interface contract will make the IDE's include it in an implementing class, so the implementer can handle the situation correctly, like ensuring it is bound or implemented as an arrow function in the class.

Like you said, `this: void` is not commonly used, but that's not necessarily a good thing. `this` in JavaScript is a fraught subject and there is certainly the need to make it clear whether a method needs to have a specific `this` instance connected to it.

As far as my initial choice, I went with the first suggestion because it seemed that it would result in clearer code when reading to have an annotation rather than change the methods to arrow functions, especially if it could have been a class type implementation.